sqlalchemylogger/handlers.py

```diff
@@ -1,42 +1,109 @@
 import logging
 import traceback
+import sqlalchemy
 from sqlalchemy import create_engine
 from sqlalchemy.orm import sessionmaker
-from .models import Log, Base
+from .models import createLogClass, Base
 from sqlalchemy.exc import OperationalError, InvalidRequestError
+from sqlalchemy_utils import database_exists, create_database
+import threading, queue, time
+from .filters import ContainsExpression, DoesNotContainExpression
+
+
+module_logs = logging.getLogger(__name__)
 
 class SQLAlchemyHandler(logging.Handler):
-    def __init__(self, sqlalchemyUrl):
+    
+    MAX_NB_LOGS = 100
+    MAX_TIMEOUT = 1
+
+    def __init__(self, sqlalchemyUrl, 
+              doesNotContainExpression = None,
+              containsExpression = None):
         super().__init__()
-        self.engine = create_engine(sqlalchemyUrl)
+        # initialize DB session
+        self.engine = create_engine(sqlalchemyUrl['url'])
+        self.Log = createLogClass(
+                  tablename = sqlalchemyUrl.get('tablename', 'logs'),
+                  tableargs = sqlalchemyUrl.get('tableargs', None))
         Base.metadata.bind = self.engine
         DBSession = sessionmaker(bind=self.engine)
         self.session = DBSession()
+        # initialize log queue
+        self.log_queue = queue.Queue()
+        # initialize a thread to process the logs Asynchronously
+        self.processor_thread = threading.Thread(target = self._processor, daemon = True)
+        self.processor_thread.start()
+        # initialize filters
+        if doesNotContainExpression:
+             self.addFilter(DoesNotContainExpression(doesNotContainExpression))
+        if containsExpression:
+             self.addFilter(ContainsExpression(containsExpression))
+
+
+    def _processor(self):
+        _terminated = False
+        while not _terminated:
+            logs = []
+            time_since_last = time.time()
+            while True:
+                try:
+                    log = self.log_queue.get(timeout=self.MAX_TIMEOUT)
+                    if log is None:
+                        # one way of killing a thread
+                        _terminated = True
+                    logs.append(log)
+                except queue.Empty:
+                     pass
+                if len(logs) > 0:
+                    if (len(logs) >= self.MAX_NB_LOGS) or (time_since_last + self.MAX_TIMEOUT <= time.time()) :
+                        self._writeLogs(logs)
+                        self.log_queue.task_done()
+                        break
+
+
+    def _writeLogs(self,logs):
```

Comment:
python methods are not camelcased: _write_logs

Reply:
ok will fix this

Reply:
fixed [here](https://github.com/danduk82/poc_oereb_logger/pull/4/commits/e6b198f9257f46965e31d6cb535e389ab68bc1e4)